src/SQLite.cs

```diff
@@ -2855,20 +2855,29 @@ public IEnumerable<T> ExecuteDeferredQuery<T> (TableMapping map)
 			var stmt = Prepare ();
 			try {
 				var cols = new TableMapping.Column[SQLite3.ColumnCount (stmt)];
+				var fastColumnSetters = new Action<T, Sqlite3Statement, int>[SQLite3.ColumnCount (stmt)];
 
 				for (int i = 0; i < cols.Length; i++) {
 					var name = SQLite3.ColumnName16 (stmt, i);
 					cols[i] = map.FindColumn (name);
+					if (cols[i] != null)
+						fastColumnSetters[i] = FastColumnSetter.GetFastSetter<T> (_conn, cols[i]);
 				}
 
 				while (SQLite3.Step (stmt) == SQLite3.Result.Row) {
 					var obj = Activator.CreateInstance (map.MappedType);
 					for (int i = 0; i < cols.Length; i++) {
 						if (cols[i] == null)
 							continue;
-						var colType = SQLite3.ColumnType (stmt, i);
-						var val = ReadCol (stmt, i, colType, cols[i].ColumnType);
-						cols[i].SetValue (obj, val);
+
```

Comment:
If a FastColumnSetter is ever not available for this column, then we don't worry about it and we just fall back to the (slow) `PropertyInfo.SetValue()` method.